pkg/volume/emptydir/empty_dir.go

```diff
@@ -528,7 +529,7 @@ func (ed *emptyDir) teardownDefault(dir string) error {
 	}
 	// Renaming the directory is not required anymore because the operation executor
 	// now handles duplicate operations on the same volume
-	return os.RemoveAll(dir)
+	return removeall.RemoveAllOneFilesystem(ed.mounter, dir)
```

Comment:
Are you sure this works as expected? [The doc](https://pkg.go.dev/k8s.io/kubernetes/pkg/util/removeall#RemoveAllOneFilesystem) says:

> It makes sure it does not cross mount boundaries, i.e. it returns an error rather than remove files from another filesystem

Does this return an error if there is a nested mount point, then? In that case, doesn't returning an error cause any issues?

Reply:
> Are you sure this works as expected?

Yes. When we use `removeall.RemoveAllOneFilesystem`, all files on the nested mount are preserved.

> Does this return an error if there is a nested mount point, then? In that case, doesn't returning an error cause any issues?

Yes. It indeed returns an error, the same way `os.RemoveAll` does.

When using `os.RemoveAll`, all the files are removed. Howerver, as it cannot remove the mount point directory, it returns the error `unlinkat /var/lib/k0s/kubelet/pods/.../volumes/kubernetes.io~empty-dir/rshared/nfs: device or resource busy`.

On the other hand, when using `removeall.RemoveAllOneFilesystem`, all files are kept and the error returned is `cannot delete directory /var/lib/k0s/kubelet/pods/.../volumes/kubernetes.io~empty-dir/rshared/nfs: it is a mount point`

Reply:
Thanks!